usnic_devinfo.c

```diff
@@ -152,7 +152,41 @@ static int devinfo(struct fi_usnic_info *finfo, int brief)
 	return 0;
 }
 
-void show_help(const char *argv0)
+static long parse_long(char *str)
+{
+	long ret;
+	char *end;
+
+	/* If usnic_ is given, then strtol will not perform conversion and this
+	 * function will return 0. Instead special case it as an error.
```

Comment:
> This comment now no longer matches the code

It does but it may not be clear. So in the option handling there is a string compare against `usnic_`, if that passes but the input string is exactly `usnic_`, then the length of the string given to `parse_long` will be 0. If this is given to `strtol`, then it will not perform any conversion and we will return `0` from this function. Once it returns we will sprintf and get `usnic_0`, and then return that device even though giving `usnic_` was probably a mistake.